### PR TITLE
VZ-4438 Create templates for all standard Kubernetes workloads

### PR DESCRIPTION
**Pull request by davisgomes**
# Description

Along with a few minor updates as testing occurs, this creates new templates for all standard Kubernetes workloads.

Fixes VZ-4438

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/webhooks/scrape_generator_webhook.go [hunk 1]
```diff
@@ -186,6 +186,7 @@ func (a *ScrapeGeneratorWebhook) createOrUpdateMetricBinding(ctx context.Context
 	// For at least deployments, the webhook is called multiple times.  The first time the UID is empty.
 	// A UID is needed for setting up the owner reference.  Return and do nothing if the UID is empty.
 	if len(unst.GetUID()) == 0 {
+		scrapeGeneratorLogger.Info("No UID found for the resource", "Namespace", unst.GetNamespace(), "Name", unst.GetName())
```
Comment (kminder) [inline application-operator/controllers/webhooks/scrape_generator_webhook.go:189]: Does this need to be info or can it be debug?

File: application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
Comment (bobdonat) [inline application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml:1]: What is the difference between the deployment, replicasets, and statefulsets templates?
If the PomtheusConfig is the same we could have a single template with a workloadSelector the handles all three.

Comment (kminder) [inline application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml:1]: Do all these need to be different files or can the workloadSelector for a single template just include them all?

Comment (kminder) [inline application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml:1]: These should be 2022 only right?

File: application-operator/internal/app/resources/workloads/hello-helidon-deployment.yaml [hunk 3]
```diff
@@ -15,6 +15,7 @@ spec:
     metadata:
       labels:
         app: hello-helidon-application
+        app.verrazzano.io/metrics-scrape: "true"
```
Comment (bobdonat) [inline application-operator/internal/app/resources/workloads/hello-helidon-deployment.yaml:18]: this is only temporary, correct?

Comment (kminder) [inline application-operator/internal/app/resources/workloads/hello-helidon-deployment.yaml:18]: Is this temporary?

File: application-operator/controllers/webhooks/scrape_generator_webhook.go (hunk 1 above)
Comment (davisgomes) [reply application-operator/controllers/webhooks/scrape_generator_webhook.go:189]: I think this is an important message if hit. It lets the user know why the metricsbinding does not get created

File: application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml (hunk 2 above)
Comment (davisgomes) [reply application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml:1]: You're right. I think I should be able to combine all of them.

File: application-operator/internal/app/resources/workloads/hello-helidon-deployment.yaml (hunk 3 above)
Comment (davisgomes) [reply application-operator/internal/app/resources/workloads/hello-helidon-deployment.yaml:18]: Yes this is a placeholder until the pod webhook creates the label

Comment (davisgomes) [reply application-operator/internal/app/resources/workloads/hello-helidon-deployment.yaml:18]: Yes. It is a placholder until the pod webhook gets created.

File: application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml (hunk 2 above)
Comment (davisgomes) [reply application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml]: Combined into a single file

Comment (davisgomes) [reply application-operator/internal/app/resources/templates/replicaset-metrics-template.yaml]: Yes, I have updated those files

**Review by kminder — APPROVED** [review]
Approved

**Pull request merged**